meilisearch-http/src/index_controller/update_actor/store/mod.rs

```diff
@@ -477,8 +478,14 @@ impl UpdateStore {
                 let _ = remove_file(path);
             });
 
-        // We don't care about the currently processing update, since it will be removed by itself
-        // once its done processing, and we can't abort a running update.
+        // If the currently processing update is from our index, we wait until it is
+        // finished before returning. This ensure that no write to the index occurs after we delete it.
+        if let State::Processing(uuid, _) = *self.state.read() {
+            if uuid == index_uuid {
+                // wait for a write lock, do nothing with it.
+                self.state.write();
```

Comment:
Where you able to ensure this line works?
I would expect rust to delete it entirely since you don't use the return value :thinking:

Reply:
Why would it do that? It has side effects